core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1854,6 +1889,11 @@ private[spark] class DAGScheduler(
               // host, including from those that we still haven't confirmed as lost due to heartbeat
               // delays.
               ignoreShuffleFileLostEpoch = isHostDecommissioned)
+
+            if (pushBasedShuffleEnabled) {
+              // Remove fetchFailed host in the shuffle push merger list for push based shuffle
+              env.blockManager.master.removeShufflePushMergerLocation(bmAddress.host)
+            }
```

Comment:
I think this would be better to be moved to `removeExecutorAndUnregisterOutputs` method right before the line `        blockManagerMaster.removeExecutor(execId)` as there it will be protected from reprocessing follow-up fetch failures. For details please see the the epoch checks. 

And you can also use `blockManagerMaster` instead of `env.blockManager.master`.